expense/activities.go

```diff
@@ -77,6 +88,41 @@ func WaitForDecisionActivity(ctx context.Context, expenseID string) (string, err
 	return "", fmt.Errorf("register callback failed status:%s", status)
 }
 
+func SignalWithStartExpenseApprovedActivity(
+	ctx context.Context,
+	expenseID string,
+	companyID int,
+) (string, error) {
+	c := ctx.Value(ClientContextKey).(client.Client)
+	workflowID := fmt.Sprintf(
+		"%s:%d",
+		"batch_payment",
+		companyID,
+	)
+	workflowOptions := client.StartWorkflowOptions{
+		ID:        workflowID,
+		TaskQueue: "expense",
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute,
+			MaximumAttempts:    5,
+		},
+	}
+	activity.GetLogger(ctx).Info("Signaling for payment", "Company ID", companyID, "Expense ID", expenseID, "WorkflowID", workflowID)
+	wr, err := c.SignalWithStartWorkflow(
```

Comment:
Never call any external APIs (including ones found on the Temporal Client) from workflows. They might break determinism by returning different results/errors on each call. 

Always use activities for this.